homeassistant/components/lastfm/sensor.py

```diff
@@ -35,95 +32,57 @@
 )
 
 
+def format_track(track: Track) -> str:
+    """Format the track."""
+    return f"{track.artist} - {track.title}"
+
+
 def setup_platform(
     hass: HomeAssistant,
     config: ConfigType,
     add_entities: AddEntitiesCallback,
     discovery_info: DiscoveryInfoType | None = None,
 ) -> None:
     """Set up the Last.fm sensor platform."""
-    api_key = config[CONF_API_KEY]
-    users = config[CONF_USERS]
-
-    lastfm_api = lastfm.LastFMNetwork(api_key=api_key)
-
+    lastfm_api = LastFMNetwork(api_key=config[CONF_API_KEY])
     entities = []
-    for username in users:
+    for username in config[CONF_USERS]:
         try:
-            lastfm_api.get_user(username).get_image()
-            entities.append(LastfmSensor(username, lastfm_api))
-        except WSError as error:
-            _LOGGER.error(error)
-            return
-
+            user = lastfm_api.get_user(username)
+            entities.append(LastFmSensor(user, lastfm_api))
+        except WSError as exc:
+            LOGGER.error("Failed to load LastFM user `%s`: %r", username, exc)
     add_entities(entities, True)
 
 
-class LastfmSensor(SensorEntity):
+class LastFmSensor(SensorEntity):
     """A class for the Last.fm account."""
 
     _attr_attribution = "Data provided by Last.fm"
     _attr_icon = "mdi:radio-fm"
 
-    def __init__(self, user, lastfm_api):
+    def __init__(self, user: User, lastfm_api: LastFMNetwork) -> None:
         """Initialize the sensor."""
-        self._unique_id = hashlib.sha256(user.encode("utf-8")).hexdigest()
-        self._user = lastfm_api.get_user(user)
-        self._name = user
-        self._lastfm = lastfm_api
-        self._state = "Not Scrobbling"
-        self._playcount = None
-        self._lastplayed = None
-        self._topplayed = None
-        self._cover = None
-
-    @property
-    def unique_id(self):
-        """Return the unique ID of the sensor."""
-        return self._unique_id
-
-    @property
-    def name(self):
-        """Return the name of the sensor."""
-        return self._name
-
-    @property
-    def native_value(self):
-        """Return the state of the sensor."""
-        return self._state
+        self._attr_unique_id = hashlib.sha256(user.name.encode("utf-8")).hexdigest()
+        self._attr_name = user.name
+        self._user = user
 
     def update(self) -> None:
         """Update device state."""
-        self._cover = self._user.get_image()
-        self._playcount = self._user.get_playcount()
-
-        if recent_tracks := self._user.get_recent_tracks(limit=2):
-            last = recent_tracks[0]
-            self._lastplayed = f"{last.track.artist} - {last.track.title}"
-
+        self._attr_entity_picture = self._user.get_image(SIZE_SMALL)
```

Comment:
Looking back, this should be a dedicated PR. It's potentially a breaking change (maybe people expect the big image)
```suggestion
        self._attr_entity_picture = self._user.get_image()
```